src/coreclr/jit/optimizer.cpp

```diff
@@ -3518,6 +3518,13 @@ bool Compiler::optNarrowTree(GenTree* tree, var_types srct, var_types dstt, Valu
 
             case GT_IND:
 
+                if (opts.OptimizationEnabled())
+                {
+                    // Don't narrow casts of indirections when optimizations are enabled
+                    // to enable more CSEs.
+                    return false;
```

Comment:
Depending on the diffs it might be good to also check what they look like if we did this opt in lowering instead

Reply:
Yeah I know, I want to check first whether diffs are equal on x64 and arm64 since CAST handling is different in lower for arm64 and x64

Reply:
Diffs: https://dev.azure.com/dnceng-public/public/_build/results?buildId=747840&view=ms.vss-build-web.run-extensions-tab not too bad as I expected. Arm64 are quite small, mostly because all loads are 4 bytes while size of the load is different on x64. I wasn't planning to finish it as I have other things to do, but if someone wants to take over - please feel free